app/src/sharedTest/java/org/oppia/android/app/profile/PinPasswordActivityTest.kt

```diff
@@ -689,6 +691,36 @@ class PinPasswordActivityTest {
     }
   }
 
+  @Test
+  fun testPinPasswordActivity_checkKeyboardType_clickShowHidePassword_keyboardTypeIsSame() {
+    ActivityScenario.launch<PinPasswordActivity>(
+      PinPasswordActivity.createPinPasswordActivityIntent(
+        context,
+        adminPin,
+        adminId
+      )
+    ).use {
+      testCoroutineDispatchers.runCurrent()
+
+      val inputType = InputType.TYPE_CLASS_NUMBER
```

Comment:
let's not check the type here explicitly. We can try accessing the input type directly from Ui

```kotlin
  @Test
  fun testPinPasswordActivity_checkInputType_clickShowHidePassword_inputTypeIsSame() {
    ActivityScenario.launch<PinPasswordActivity>(
      PinPasswordActivity.createPinPasswordActivityIntent(
        context,
        adminPin,
        adminId
      )
    ).use {
      testCoroutineDispatchers.runCurrent()

      var inputType: Int = 0
      it.onActivity {
        inputType = it.findViewById<PinView>(R.id.input_pin).inputType
      }
      onView(withId(R.id.input_pin))
        .check(matches(withInputType(inputType)))
      onView(withId(R.id.show_pin)).perform(click())
      onView(withId(R.id.input_pin))
        .check(matches(withInputType(inputType)))
    }
  }
```

Reply:
let me know if this test case is passing on robolectric too, then we can go with this.

Reply:
Doing it this way makes more sense. Actually I'm not that familiar with ActivityScenario so I wasn't able to find a way to extract the inputType at runtime.

Reply:
This test passes on robolectric